deepchem/data/datasets.py

```diff
@@ -1607,10 +1607,18 @@ def iterate(dataset: DiskDataset, batch_size: Optional[int], epochs: int):
       # mp.dummy aliases ThreadPool to Pool
       pool = Pool(1)
 
+      import torch.distributed as dist
+
+      if dist.is_initialized():
+        world_size = dist.get_world_size()
+      else:
+        world_size = 1
+
       if batch_size is None:
         num_global_batches = num_shards
       else:
-        num_global_batches = math.ceil(dataset.get_shape()[0][0] / batch_size)
+        # num_global_batches = math.ceil(dataset.get_shape()[0][0] / batch_size)
+        num_global_batches = (dataset.get_shape()[0][0] // world_size) // batch_size
```

Comment:
Why did you remove the `math.ceil()` and change `/` to `//`?  When you have a final partial batch, that will cause it to be skipped.